src/components/FormGroup/FormGroup.tsx

```diff
@@ -0,0 +1,178 @@
+import { Theme, useTheme } from '../../hooks/useTheme'
+import { StatusLabel } from '../StatusLabel'
+import { Heading, HeadingTypes } from '../Heading'
+import { FaExclamationCircleIcon } from '../Icon'
+
+type innerMarginType = 'XXS' | 'XS' | 'S'
+type Props = {
+  label: string
+  labelType?: HeadingTypes
+  labelId?: string
+  innerMargin?: innerMarginType
+  statusLabels?: Array<ComponentProps<typeof StatusLabel>>
+  helpMessage?: ReactNode
+  errorMessages?: string | string[]
+  children: ReactNode
+  disabled?: boolean
+  className?: string
+}
+
+export const FormGroup: FC<Props> = ({
+  label,
+  labelType = 'blockTitle',
+  labelId,
+  innerMargin = 'XS',
+  statusLabels,
+  helpMessage,
+  errorMessages,
+  children,
+  disabled,
+  className,
+}) => {
+  const theme = useTheme()
+  const disabledClass = disabled ? 'disabled' : ''
+
+  return (
+    <Wrapper className={`${className} ${disabledClass}`} themes={theme}>
+      <Label themes={theme} id={labelId} margin={innerMargin}>
+        <TitleWrapper>
+          <Title type={labelType} disabled={disabled}>
+            {label}
+          </Title>
+          {statusLabels && (
+            <StatusLabels themes={theme}>
+              {statusLabels.map((StatusLabelItem, index) => (
+                <LabelItem
+                  key={index}
+                  type={StatusLabelItem.type}
+                  className={StatusLabelItem.className}
+                  themes={theme}
+                >
+                  {StatusLabelItem.children}
+                </LabelItem>
+              ))}
+            </StatusLabels>
+          )}
+        </TitleWrapper>
+
+        {helpMessage && <HelpMessage themes={theme}>{helpMessage}</HelpMessage>}
+
+        {errorMessages &&
+          (typeof errorMessages === 'string' ? [errorMessages] : errorMessages).map(
+            (message, index) => (
+              <ErrorMessage themes={theme} key={index}>
+                <ErrorIcon
+                  color={disabled ? theme.palette.TEXT_DISABLED : theme.palette.DANGER}
+                  themes={theme}
+                  size={14}
+                />
+                <ErrorText>{message}</ErrorText>
+              </ErrorMessage>
+            ),
+          )}
+      </Label>
+      <Body>{children}</Body>
+    </Wrapper>
+  )
+}
+
+const Wrapper = styled.div<{ themes: Theme }>`
+  ${({ themes }) => {
+    const { palette } = themes
+    return css`
+      display: block;
+
+      &.disabled {
+        color: ${palette.TEXT_DISABLED};
+      }
+    `
+  }}
+`
+
+const Label = styled.label<{ themes: Theme; margin: innerMarginType }>`
+  ${({ themes, margin }) => {
+    const { size } = themes
+
+    return css`
+      display: block;
+      margin-bottom: ${size.pxToRem(size.space[margin])};
+    `
+  }}
+`
+
+const TitleWrapper = styled.div`
+  display: flex;
+  align-items: baseline;
+  flex-wrap: wrap;
+`
+
+const Title = styled(Heading)`
+  display: inline-block;
+`
```

Comment:
[IMO] こちら、`Heading` コンポーネント側に `disabled` オプションを追加するのではなく、ここで Extending styles すれば良いのでは？と思いました。

```suggestion
const Title = styled(Heading)<{ themes: Theme }>`
  display: inline-block;
  &.disabled {
    color: ${({ themes }) => themes.palette.TEXT_DISABLED};
  }
`
```

Reply:
おっ、👆で style 当てられます？ 🤔 

割と sytled-components の Extending styles ふんいきでやっていてアレなんですが、Wrapper の各種 color の style の方が詳細度が高いと思っていて外から当てるのは難しいと思っていました！

https://github.com/kufu/smarthr-ui/blob/master/src/components/Heading/Heading.tsx#L48

Reply:
Suggested change の例だと `Title` に `className={disabled ? 'disabled' : undefined}` を設定すれば style は当たるようです。
この場合では、`&.disabled` と `Title` 側の `&.blockTitle` の詳細度は同じになっていますが、Extending styles した `&.disabled` が後勝ちしているようです。

ちなみに Extending styles 側が詳細度で負ける場合は `&&.disabled` といった感じで `&` の数で詳細度を上げてやっても良いかもしれません。

Reply:
>ちなみに Extending styles 側が詳細度で負ける場合は &&.disabled といった感じで & の数で詳細度を上げてやっても良いかもしれません

なるほど…！

`.hoge.hoge.disabled` みたいな感じになるんですね。
そもそもですが、同じクラス重ねて詳細度あげることができるのね… CSS 力…

Reply:
wmoai san のコメントの通りで無事適用できたので修正しました！
Header コンポーネントへの修正は revert してますー

Reverted in 91044b04b6e7bc51cb2fb1cb1a2afab736df8575
Fixed in b167fb7280507eb66682fc01f24f695194ae2336

Reply:
:bow: :bow: :bow: